xarray/core/common.py

```diff
@@ -874,6 +874,68 @@ def rolling_exp(
 
         return rolling_exp.RollingExp(self, window, window_type)
 
+    def shuffle_by(
```

Comment:
Alternatively, we could provide `Dataset.groupby().shuffle_object()` for example.